velox/benchmarks/basic/LikeFunctionsBenchmark.cpp

```diff
@@ -0,0 +1,202 @@
+        auto fixedPatternLength =
+            std::min(vector_size_t(inputString.size()), 10);
+        auto fixedPatternString = inputString.substr(0, fixedPatternLength);
+        return fixedPatternString + generateRandomString(kAnyWildcardCharacter);
+      }
+      case PatternKind::kSuffix: {
+        auto fixedPatternStartIdx =
+            std::max(vector_size_t(inputString.size() - 10), 0);
+        auto fixedPatternString = inputString.substr(fixedPatternStartIdx, 10);
+        return generateRandomString(kAnyWildcardCharacter) + fixedPatternString;
+      }
+      default:
+        return inputString;
+    }
+  }
+
+  size_t run(VectorPtr input, const StringView patternString) {
+    const auto data = makeRowVector({input});
+    size_t cnt = 0;
+    auto result = FunctionBaseTest::evaluate<SimpleVector<bool>>(
+        fmt::format("like(c0, '{}')", patternString), data);
+    cnt += result->size();
+    folly::doNotOptimizeAway(cnt);
+
+    return cnt;
+  }
+
+  size_t run(PatternKind patternKind) {
+    const auto input = inputFuzzer_->values()->as<StringView>();
+    auto patternString = generatePattern(patternKind, input[0].str());
+    std::vector<std::string> patternVector(FLAGS_vector_size, patternString);
+    const auto data = makeRowVector({inputFuzzer_});
+    size_t cnt = 0;
+
+    auto result = FunctionBaseTest::evaluate<SimpleVector<bool>>(
+        fmt::format("like(c0, '{}')", patternString), data);
+    cnt += result->size();
+    folly::doNotOptimizeAway(cnt);
+
+    return cnt;
+  }
+
+  // We inherit from FunctionBaseTest so that we can get access to the helpers
+  // it defines, but since it is supposed to be a test fixture TestBody() is
+  // declared pure virtual.  We must provide an implementation here.
+  void TestBody() override {}
+
+ private:
+  static constexpr const char* kWildcardCharacterSet = "%_";
+  static constexpr const char* kAnyWildcardCharacter = "%";
+  VectorPtr inputFuzzer_;
+};
+
+std::unique_ptr<LikeFunctionsBenchmark> benchmark;
+
+BENCHMARK_MULTI(wildcardExactlyN) {
+  return benchmark->run(facebook::velox::functions::PatternKind::kExactlyN);
+}
+
+BENCHMARK_MULTI(wildcardAtLeastN) {
+  return benchmark->run(PatternKind::kAtLeastN);
+}
+
+BENCHMARK_MULTI(fixedPattern) {
+  return benchmark->run(PatternKind::kFixed);
+}
+
+BENCHMARK_MULTI(prefixPattern) {
+  return benchmark->run(PatternKind::kPrefix);
+}
+
+BENCHMARK_MULTI(suffixPattern) {
+  return benchmark->run(PatternKind::kSuffix);
+}
+
+BENCHMARK_DRAW_LINE();
+
+BENCHMARK_MULTI(tpchQuery2) {
+  auto tpchPart = genTpchPart(FLAGS_num_rows);
+  auto partTypeVector = tpchPart->childAt(4);
+  return benchmark->run(partTypeVector, "%BRASS");
+}
+
+BENCHMARK_MULTI(tpchQuery9) {
+  auto tpchPart = genTpchPart(FLAGS_num_rows);
+  auto partNameVector = tpchPart->childAt(1);
+  return benchmark->run(partNameVector, "%green%");
+}
+
+BENCHMARK_MULTI(tpchQuery13) {
+  auto tpchOrders = genTpchOrders(FLAGS_num_rows);
```

Comment:
if genTpchOrders is not trivial (runtime wise) use suspender. or move it inside run and call it in the suspender block there to avoid have two suspensions